app/src/main/java/org/mozilla/fenix/collections/CollectionCreationTabListAdapter.kt

```diff
@@ -44,10 +44,8 @@ class CollectionCreationTabListAdapter(
             when (payloads[0]) {
                 is CheckChanged -> {
                     val checkChanged = payloads[0] as CheckChanged
-                    if (checkChanged.shouldBeChecked) {
-                        binding.tabSelectedCheckbox.isChecked = true
-                    } else if (checkChanged.shouldBeUnchecked) {
-                        binding.tabSelectedCheckbox.isChecked = false
+                    if (checkChanged.shouldBeChecked || checkChanged.shouldBeUnchecked) {
+                        holder.changeCheckBoxState(checkChanged.shouldBeChecked)
```

Comment:
This check assumes that whenever `shouldBeUnchecked` is true `shouldBeChecked` is false.
Which sounds right as per the naming but the current if/else does not make this assumption and makes me wonder why we had this abstraction in the first place and whether we can just drop the `shouldBeUnchecked` property or it's an important edgecase we need to still acount for.

Reply:
@Robert-Aron293 to minimize the risk of regressions let's keep the previous if/else approach for now and I'll investigate how `CheckChanged` is used